Swagger - update API Documentation: capping support

padv2010 opened the pull request:
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDLP-3108

### Description

Documents API changes after adding Capping support.

cwrw commented on docs/source/release-notes.html.md:
```diff
@@ -7,6 +7,24 @@ weight: 8
+
+This is because from the 2024/25 academic year onwards there'll be a set maximum number of places each provider can offer per NPQ that DfE will pay for.
+
+Providers using all versions of the API can set the new funded_place field in the 'Accept an application' request body to true or false. They will also see the funded_place field in the following endpoint response bodies:
```
```suggestion
Providers using all versions of the API can set the new `funded_place` field in the 'Accept an application' request body to true or false. They will also see the `funded_place` field in the following endpoint response bodies:
```

cwrw commented on swagger/v3/api_spec.json:
```diff
@@ -161,12 +161,12 @@
         }
       }
     },
-    "/api/v3/partnerships/ecf": {
```
could we avoid running swaggerise on v3, and amending manually, as this is the only version that has not been fixed properly, and will change up the whole file?

cwrw replied:
can we please avoid this large change, can we jump on a call?

leandroalemao reviewed (changes requested):
Hi @padv2010 there're new fields added on v1/v2/v3 `npq_participant_serializer.rb` so some extra changes in the related docs are needed as well ie: https://github.com/DFE-Digital/early-careers-framework/blob/main/app/serializers/api/v1/npq_participant_serializer.rb#L26

padv2010 commented:
Thanks for the review @cwrw @leandroalemao 

@cwrw I have addressed both changes, 

@leandroalemao, I have updated v1 for ParticipantProfile, V2 and V3, unless I am missing something, are already there. I would appreciate some help though, because I can't add a new Schema object to V1 to support `funded_places`

padv2010 commented on docs/source/release-notes.html.md:
```diff
@@ -7,6 +7,25 @@ weight: 8
 
 If you have any questions or comments about these notes, please contact DfE via Slack or email.
 
+## ?? (pending sandbox release)
```
```suggestion
## 14 June 2024 (pending sandbox release)
```

cwrw commented on docs/source/release-notes.html.md:
```diff
@@ -7,6 +7,25 @@ weight: 8
+
+Providers using all versions of the API can set the new `funded_place` field in the 'Accept an application' request body to true or false. They will also see the `funded_place` field in the following endpoint response bodies:
+
+'View all applications'
+
+'View a specific application'
+
+'View all participant data'
+
+'View a single participant's data'
```
```suggestion
- 'View all applications'

- 'View a specific application'

- 'View all participant data'

- 'View a single participant's data'
```

cwrw reviewed (commented):
@padv2010 thank you for addressing comments. I think v3 is still missing a few places, let's chat tomorrow please. Thanks

cwrw commented on swagger/v3/component_schemas/NPQApplicationAcceptAttributesRequest.yml:
```diff
@@ -1,22 +1,7 @@
 description: "The NPQ Application acceptance request"
 type: object
-required:
```
Can we keep those please? as those are required for the acceptance on v3? and add the new funded place

cwrw commented on swagger/v3/component_schemas/NPQApplicationAcceptRequest.yml:
```diff
@@ -7,4 +7,4 @@ example:
   data:
     type: "npq-application-accept"
     attributes:
-      schedule_identifier: "npq-leadership-spring"
```
same here

cwrw reviewed (approved):
thank you @padv2010

leandroalemao reviewed (approved):
tks @padv2010 🔝
